### ISSUE TITLE
Newer versions of schematron with report element?

### ISSUE THREAD

**Issue by baerrach** [phax/ph-schematron#22]
http://www.schematron.com/iso/P8.html

5.4.11 report element

My google-fu is failing me and I can not work out where you got your iso-schematron-xslt2 zip files from and whether there is a newer one with the new elements.

Hopefully you know how to find those!

**Comment by phax** [comment]
I'm pretty sure I downloaded them from www.schematron.com but I can't find them anymore :(
Anyway I just noticed that a new version of the standard came out - http://standards.iso.org/ittf/PubliclyAvailableStandards/c055982_ISO_IEC_19757-3_2016.zip - I don't support this stuff at the moment.....

**Comment by phax** [comment]
The `report` element is not something new.
Schematron has built-in "pre-processing" that converts `report` elements to `assert` elements by negating the condition.

**Comment by baerrach** [comment]
Then I must be doing something wrong.

I get 

Failed to convert 'path\to\file.sch': the Schematron resource is invalid (com.helger.maven:ph-sch2xslt-maven-plugin:4.0.1-SNAPSHOT:convert:default:generate-resources)

Which nothing useful in the error message telling me why it doesn't work.

It also doesn't like putting in ns elements either.

I suppose I should look at running this via the command line to compare outputs.

**Comment by phax** [comment]
I found http://www.schematron.com/implementation.html where I grabbed the XSLTs from - this page is heavily unmaintained :(

Are you able to paste the Schematron here? Or want to send me a PM?

**Comment by baerrach** [comment]
I found the problem and provided the fix via a new pull request.

When I ran it on the command line you get the full stack trace - which clearly marked where the problem was.

m2e integraion doesn't want the linked exception, as that wraps the cause. Instead it needs to pull the cause out and use its message string. As I was displaying the wrapped exception I was getting a generic error that didn't help me diagnose the problem.

Additionally when it there are failures, the problem view will also contain an error for pom.xml at the line the ph-sch2xslt-maven-plugin is defined with

```
Failed to convert 'D:\ide\ira-schematron\ira-validation\src\main\schematron\ebm.sch': the Schematron resource is invalid (com.helger.maven:ph-sch2xslt-maven-plugin:4.0.1-SNAPSHOT:convert:default:generate-resources)
```

I'm not sure how to resolve that. The plugin needs to through an error to signal failure.
